Cancel mining if blockchain's tip index is changed (#517)
planetarium/libplanet: CHANGES.md

### Backward-incompatible interface changes

>  -  `BlockChain<T>.MineBlock()` is now async and became to throw

Reviewer: ```suggestion
 -  `BlockChain<T>.MineBlock()` is now `async` and became to throw
```

### Added interfaces

>  -  Users can give a cancellation token to `Block<T>.Mine()` and `Hashcash.Answer()`

Reviewer: ```suggestion
 -  Users became able to give a cancellation token to `Block<T>.Mine()` and
```

>     to cancel the operation.  [[#460], [#517]]

Reviewer: ```suggestion
    `Hashcash.Answer()` to cancel the operation.  [[#460], [#517]]
```

>     `Hashcash.Answer()` to cancel the operation.  [[#460], [#517]]

Reviewer: Although the added parameter has a default value, this makes a bytecode-level API (i.e., ABI) change so that library users need to recompile their code.  This should go to backward-incompatible interface changes.

 -  Added `BlockChain<T>.AddTipHandler()` and `BlockChain<T>.RemoveTipHandler()`
    which take `EventHandler` with `long` typed tip index argument that invoked when

>     `BlockChain<T>`'s tip is changed.  [[#517]]

Reviewer: ```suggestion
    `BlockChain<T>.Tip` is changed.  [[#517]]
```
